src/main/java/org/zone/region/flag/entity/player/move/preventing/PreventPlayersListener.java

```diff
@@ -53,4 +58,142 @@ public void onPlayerMove(MoveEntityEvent event, @Getter("entity") Player player)
         event.setCancelled(true);
     }
 
+    @Listener
+    public void onPlayerLoginEvent(ServerSideConnectionEvent.Join joinevent) {
+        ServerPlayer serverPlayer = joinevent.player();
+        Optional<Zone> opZone = ZonePlugin
+                .getZonesPlugin()
+                .getZoneManager()
+                .getPriorityZone(joinevent.player().world(), joinevent.player().position());
+
+        if (opZone.isEmpty()) {
+            return;
+        }
+
+        if (!(opZone.get().containsFlag(FlagTypes.PREVENT_PLAYERS))) {
+            return;
+        }
+
+        Optional<PreventPlayersFlag> flag = opZone
+                .get()
+                .getFlag(FlagTypes.PREVENT_PLAYERS);
+        if (flag.isEmpty()) {
+            return;
+        }
+        if (flag.get().hasPermission(opZone.get(),serverPlayer.uniqueId())) {
+            Optional<Vector3i> opTeleportPos = opZone
+                    .get()
+                    .getRegion()
+                    .getTrueChildren()
+                    .stream()
+                    .filter(boundedRegion -> {
+                        Vector3i position = boundedRegion.getMin().add(-1, 0, -1);
+                        position = serverPlayer
+                                .world()
+                                .highestPositionAt(position);
+                        final Vector3i finalPosition = position;
+                        //check if in any zone
+                        return !ZonePlugin
+                                .getZonesPlugin()
+                                .getZoneManager()
+                                .getZones()
+                                .stream()
+                                .anyMatch(zone -> zone.inRegion(serverPlayer.world() ,
+                                                                finalPosition.toDouble()));
+                    })
+                    .findAny()
+                    .map(boundedRegion -> {
+                        Vector3i position = boundedRegion.getMin().add(-1, 0, -1);
+                        return serverPlayer
+                                .world()
+                                .highestPositionAt(position);
+                    });
+            if (opTeleportPos.isPresent()) {
+                serverPlayer.setPosition(opTeleportPos.get().toDouble());
+            }else {
+                serverPlayer.setPosition(serverPlayer.world().properties().spawnPosition().toDouble());
+            }
+        }else {
+            return;
```

Comment:
```java
} else {
    return
}
```

When its the last line, is the same as just ``}``

Reply:
mmm

Reply:
but is this needed/

Reply:
Thats the point that i just said. Its not.